integr8ly/integreatly-operator
# copy and link pull secret

https://issues.redhat.com/browse/DEL-43

When installing branches created by the delorean ews pipelines, the serviceAccount pulling the image must have the requisite imagePullSecret to successfully pull the image. This PR alters the namespace reconciliation so that:

- If there is a secret defined in the installation spec that secret is copied to each namespace created by the operator
- For each SA in that namespace the secret is added as an imagePullSecret 

*Verification*

1. make cluster/prepare
2. make cluster/prepare/crd
3. create secret in rhmi-redhat-operator namespace
4. add this line to installation cr `pullSecret: {"name": "<secret-name>","namespace":"rhi-redhat-operator"}`
5. make code/run/service_account
6. verify that:

1. The secret specified in the cr has been copied to each namespace created by the operator
2. The secret is in the imagePullSecrets array in every serviceAccount in each operator created namespace
3. A syndesis-pull-secret exists in the fuse namespace
4. A `samples-registry-credentials` secret exists in the 3scale namespace
5. An `apicurito-pull-secret` exists in the apicurito and apicurito-operator namespaces
